pvfactors/irradiance/models.py

```diff
@@ -67,12 +75,25 @@ def fit(self, timestamps, DNI, DHI, solar_zenith, solar_azimuth,
             solar_azimuth = np.array([solar_azimuth])
             surface_tilt = np.array([surface_tilt])
             surface_azimuth = np.array([surface_azimuth])
+            if GHI is not None:
+                GHI = np.array([GHI])
         # Length of arrays
         n = len(DNI)
         # Make sure that albedo is a vector
         if np.isscalar(albedo):
             albedo = albedo * np.ones(n)
 
+        # Save and calculate total POA values from Perez model
+        if GHI is None:
+            # Calculate GHI if not specified
+            GHI = DNI * cosd(solar_zenith) + DHI
+        self.GHI = GHI
+        self.DHI = DHI
+        perez_front_pvrow = get_total_irradiance(
+            surface_tilt, surface_azimuth, solar_zenith, solar_azimuth,
+            DNI, GHI, DHI, albedo=albedo)
+        total_perez_front_pvrow = perez_front_pvrow['poa_global']
```

Comment:
Any reason to not add it to `self.total_perez['front_pvrow']` directly?

Reply:
good catch!